woocommerce/woocommerce-gateway-stripe
# Improve error message when payment method creation fails

Partly addresses #3558 
Fixes #3571 

## Changes proposed in this Pull Request:
In classic (shortcode) checkout, when the customer does not provide a billing country when paying via credit card, payment method creation fails and the raw error from Stripe is displayed.

In this PR, we catch that error and output a more user-friendly error message.

## Testing instructions
1. Go to WooCommerce > General and set "Default customer location" to "No location by default".
2. As a customer, clear your cookies to clear out previously entered billing addresses.
3. Add a product to your cart and go to **shortcode** checkout.
4. Without entering anything else, enter your test credit card info and submit.
5. In `develop`, you will see the raw error from Stripe. In this branch, you will see the user-friendly error message.

**BEFORE**
<img width="300" alt="Screenshot 2024-11-05 at 10 30 08 AM" src="https://github.com/user-attachments/assets/f56d624e-c6d3-45a5-98e9-4dc5eee04779">

**AFTER**
<img width="300" alt="Screenshot 2024-11-05 at 11 00 42 AM" src="https://github.com/user-attachments/assets/b23d584e-9316-4843-931b-ab52212f571e">



---


**Post merge**
